### ISSUE TITLE
importlib.resources.files does not implement __div__

### ISSUE THREAD

**Issue by iamkroot** [Nuitka/Nuitka#1997]
Continuing #1451, I'm seeing an error with `importlib.resources.files` on Python 3.10 when trying standalone build.

Specifically, it comes up when packaging a dependency: [desktop_notifier](https://github.com/samschott/desktop-notifier/). These are the relevant lines:

https://github.com/samschott/desktop-notifier/blob/c103d8fc4/src/desktop_notifier/base.py#L25-L28 

```python
from importlib.resources import as_file, files  # type:ignore

def resource_path(package: str, resource: str) -> ContextManager[pathlib.Path]:
    return as_file(files(package) / resource)
```

On executing the built binary, this is the error-
```
TypeError

  unsupported operand type(s) for /: 'nuitka_resource_reader_files' and 'str'

  at __main__.dist/desktop_notifier/base.py:28 in resource_path
```

Looks like `nuitka_resource_reader_files` class doesn't have a override for `/` op.

* Nuitka version, full Python version, flavor, OS, etc. as output by *this*
  command.

  > python -m nuitka --version
```
1.3.7
Commercial: None
Python: 3.10.8 (main, Nov  1 2022, 14:18:21) [GCC 12.2.0]
Flavor: Unknown
Executable: /home/kroot/.virtualenvs/trakt-scrobbler-EQl8GXq9-py3.10/bin/python
OS: Linux
Arch: x86_64
Distribution: Manjaro (based on arch) None
Version C compiler: /usr/bin/gcc (gcc).
```
* How did you install Nuitka and Python?
 Installed in virtualenv using pip.

* CLI command (trying to package https://github.com/iamkroot/trakt-scrobbler/):
```
python -m nuitka trakt_scrobbler/__main__.py --standalone --include-data-files=trakt_scrobbler/config_default.yaml=./config_default.yaml --include-package-data="desktop_notifier,trakt_scrobbler" --include-data-dir=/home/kroot/.virtualenvs/trakt-scrobbler-EQl8GXq9-py3.10/lib/python3.10/site-packages/desktop_notifier/resources=desktop_notifier/resources --include-module=desktop_notifier.resources
```
(I am sure some of these are redundant/unnecessary; will remove them later)

**Comment by iamkroot** [comment]
Ah, looking at
 https://github.com/Nuitka/Nuitka/blob/9c815255aae6b2af9ae6851bc091d97d09fb332a/nuitka/build/static_src/MetaPathBasedLoaderResourceReaderFiles.c#L304

it seems Nuitka _does_ implement `__truediv__`; but it doesn't handle strings?

Update 2:
I manually edited the code to use `joinpath` directly instead of `/`.
```python
from importlib.resources import as_file, files  # type:ignore

def resource_path(package: str, resource: str) -> ContextManager[pathlib.Path]:
    return as_file(files(package).joinpath(resource))
```

Now I'm getting a different error. The result has two slashes for some reason-
`/tmp/tmpcbjf66hz/REDACTED/__main__.dist/desktop_notifier/resources//python.png`

call: `resource_path("desktop_notifier.resources", "python.png")`

Is the implementation of `joinpath` incorrect somehow?

**Comment by kayhayen** [comment]
As an aside, you want to use `--include-package-data` which doesn't use file paths and therefore is much easier to use. You must be coming from a less capable tool there.

Also you probably want `--python-flag=-m` to have a proper binary name and compatibility.

The `joinpath` and `/` are internally the same, more or less, it calls the same code. A different error is therefore very much unexpected.

**Comment by kayhayen** [comment]
I think, this is `files` on a different object, it's strange this would not work.

I have developed this using this example:

```
import importlib.resources
import inspect

import jsonschema

print(importlib.resources.files("jsonschema"))

print(
    inspect.getargs(
        importlib.resources.files("jsonschema").joinpath("lala").open.__code__
    )
)
try:
    print(importlib.resources.files("jsonschema").joinpath("schemas/draft3.json").open("r"))
except OSError as e:
    print(e)
```

Maybe `as_file` is trying funny things?

**Comment by kayhayen** [comment]
Bah, this is the variant that forces the embedded data files of Nuitka to temp files for no good reason. There also is no reason to use it over `.open` in the first place. That said, it works perfectly fine for me. So, this is happing how. Maybe you are asking it about a package not included. I guess, this again comes down to not having a minimal reproducer. Can you provide one?

**Comment by kayhayen** [comment]
Ah, this is more like `.name` it seems, which gives the filename.

**Comment by kayhayen** [comment]
So, it seems something in 3.10 makes it such, that it becomes necessary to register with `as_file` where 3.9 just worked, even without a temporary file being invoked. I am not even sure why it works there. But I kind of assumed these versions were the same. I even saw the `TypeError` you had, although it totally makes no sense, since that method is of course implemented.

**Comment by kayhayen** [comment]
So, with a proper registry, I can make it call our own `as_file` which is just the iterable files object. This however will have to provide an `__enter__`/`__exit__` implementation now, since you can use these in a with statement apparently. This has been inconsistent in 3.9, and I think I now recall that the fallback code didn't work, and had to be avoided with anti-bloat work, because that actually was a bug of 3.9 to not do it for the fallback.

My changes now register things, so they will not go through the fallback that would create a temporary file copy, which will improve 3.9 as well.

Adding the context manager interface will be trivial as well. Lets see what that changes.

**Comment by kayhayen** [comment]
Thanks for your report, this is fixed on the factory branch, which is a development version under rapid development. You can try it out by going here: https://nuitka.net/doc/factory.html 

Feedback if this is working is very welcome, just please do not share plans of doing it, but rather confirmations or denials of it working.

**Comment by kayhayen** [comment]
This has been released as part of 1.4 already.
